sprint-mission-2/src/main/java/com/sprint/mission/discodeit/service/file/FileUserService.java

```diff
@@ -0,0 +1,65 @@
+package com.sprint.mission.discodeit.service.file;
+
+import com.sprint.mission.discodeit.entity.User;
+import com.sprint.mission.discodeit.service.UserService;
+
+import java.io.Serializable;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+public class FileUserService implements UserService {
+    private static final Path directory = Paths.get("/Users/mac/IdeaProjects/6-sprint-mission/sprint-mission-2/src/main/resources/UserDate/");
+    private static final FileEdit instance = new FileEdit();;
+
+    //클래스 외부에서 접근할 필용가 없는 메소드라서 플라이비 함
+    //객체 UUID를 파일 이름으로 정함
+    private Path filePath(User user){
+        return directory.resolve(user.getUserId().toString().concat(".ser"));
+    };
+
+    public FileUserService() {
+        instance.init(directory);
+    }
+
+    //사용자 별로 다른 파일? 아니면 같은 파일?
+    //일단은 같은 파일 진행
+
+    public User createUser(String username, int age , String email){
+        User newUser = new User(username, age, email);
+        instance.save(filePath(newUser),newUser);
+        return newUser;
+    };
+    public User findUserById(UUID id){
+        List<User> users = instance.load(directory);
+        return users.stream()
+                .filter(user -> user.getUserId().equals(id))
+                .toList()
+                .get(0);
+    };
+    public User findUserByUserEmail(String userEmail){
```

Comment:
아래와 같이 메서드명 바꿔도 동일한 의미를 전달할 수 있을 것 같아요.
```suggestion
    public User findUserByEmail(String userEmail){
```

Reply:
넵!